docs/oob-s3-ring.md

```diff
@@ -0,0 +1,165 @@
+  the ingestion workflow.
+* The ingestion process is composed of two phases:
+    - **Initial ingestion of existing objects**
+    - **Out of band updates for new objects**
+
+#### Initial ingestion of existing objects
+
+* Record the latest sequence ID from the Metadata Server (send a query using the
+  `metadataWrapper`) that is part of the S3 connector - this will serve as a
+  marker that will be stored as a `log offset` to bookmark the point in journal
+  to resume later, after the initial ingestion is completed.
+* Send List Object Versions API request to S3C to receive a list of
+  object versions in the bucket with details like  `key` and `versionId`.
+* Retrieve object JSON from S3C Metadata with the `key` and `versionId`.
+* `Next Continuation Token` from listings is used to keep track of listing, to
+  recover during unexpected pod/node failures
+
+**Note: Requests to the remote S3C Metadata are performed over S3C's Cloudserver
+which proxies the request to Metadata. The request is secure and signed by
+AWS V4 authentication scheme.**
+
+#### Out of band updates for new objects
+
+After initial ingestion is completed, Ingestion service will use it's producer
+to read from the remote S3C's Metadata and queue the journal entries to Kafka.
+Ingestion consumer picks up the entry from Kafka and processes the following
+fields for object metadata:
+
+* `owner`
+* `ownerDisplayName`
+* `owner-id`
+* `location` if defined
+* `dataStoreName` should match the S3C location name
+
+The values for `owner`, `ownerDisplayName`, and `owner-id` are changed to match
+the Orbit user information to allow `GET` operations.
+The `location` and `dataStoreName` are changed to allow `PUT` and `DELETE`
+operations by the Orbit user.
+
+A Zenko deployment will have 1 `ingestion-populator` pod and multiple
+`ingestion-processor` pods. The number of ingestion processor pods depends on
+the `nodeCount` value specified during deployment (default is 3). The
+`ingestion-populator` pod can support multiple ingestion workflows. The
+logReader for each ingestion populator will be setup to source logs for the
+specified ingestion bucket.
+
+## Dependencies
+
+* RING with S3 Connector on version 7.4.2 or newer
+* Backbeat
+* MongoDB
+* Kafka
+* Zookeeper
+
+## Bucket/Object-Key Parity
+
+The ingestion source is a bucket on RING registered as a storage location with
+`bucketMatch` enabled. This ensures an exact match of object names without any
+prefixes (unlike the prefix schema used in Zenko locations that host multiple
+Zenko 'buckets').
+
+## Avoiding Retro Propagation
+
+The S3C location in `mirror mode` is added as a location constraint to the
+Zenko bucket. This allows putting objects in the Zenko bucket, with object's
+metadata stored in Zenko, and object's metadata & data stored in S3C.
+To avoid re-ingesting these objects back into Zenko, objects will be assigned
+`scal-source: zenko-<instance id>` in their in their user-metadata. Any object
```

Comment:
'in their' twice here.

As a note: Is Avoiding Retro Propagation, the prevention against considering a PUT through Zenko as an out of band update? right?

Reply:
> As a note: Is Avoiding Retro Propagation, the prevention against considering a PUT through Zenko as an out of band update? right?

Yup, that's correct